Cython/Compiler/ExprNodes.py

```diff
@@ -11441,11 +11441,19 @@ def generate_result_code(self, code):
         ))
         code.put_gotref(shapes_temp, py_object_type)
 
+        code.putln("#if CYTHON_COMPILING_IN_LIMITED_API")
+        code.putln('%s = __pyx_array_new(%s, %s, PyBytes_AsString(%s), "%s", (char *) %s); %s' % (
```

Comment:
Lacks error handling. I'd rather extract *only* the `char*` reading and keep everything else as before, and then only add error handling for this call in the Limited API case.

Reply:
I've updated this per what I think you were looking for, but please let me know if I've missed something.